File: Creational/Lab4/Python/estate_to_buy_factory.py
from estate_factory import *

class EstateToBuyFactory(EstateFactory):
    def create_estate(self, type: str) -> RealEstate:
        e = self.estate_from_type(type)
        e.is_buyable = True
        e.is_rentable = False
        e.is_sellable = False
        return e

File: Creational/Lab4/C#/Factories/EstateToBuyFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_.Factories
{
    public class EstateToBuyFactory : EstateFactory
    {
        public override RealEstate CreateEstate(string type)
        {
            var e = EstateFromType(type);
            e.IsBuyable = true;
            e.IsRentable = false;
            e.IsSellable = false;
            return e;
        }
    }
}
